pokemonred_puffer/environment.py

```diff
@@ -809,7 +809,11 @@ def update_seen_coords(self):
         x_pos, y_pos, map_n = self.get_game_coords()
         self.seen_coords[(x_pos, y_pos, map_n)] = 1
         self.seen_coords_since_blackout.add((x_pos, y_pos, map_n))
-        self.explore_map[local_to_global(y_pos, x_pos, map_n)] = 1
+        try:
+            self.explore_map[local_to_global(y_pos, x_pos, map_n)] = 1
+        except IndexError:
+            gy, gx = local_to_global(y_pos, x_pos, map_n)
+            print(f"coord out of bounds! global: ({gx}, {gy}) game: ({x_pos}, {y_pos}, {map_n})")
```

Comment:
```suggestion
    self.explore_map[local_to_global(y_pos, x_pos, map_n)] = 1
```

looks like this is handled by the local_to_global function. But if you really want to do something like this, I'd change local_to_global to return an `Optional[tuple[int, int]]` and change this section to read like:

```python
if rc := local_to_global(y_pos, x_pos, map_n):
    self.explore_map[rc[0], rc[1]] = 1
```

Reply:
This is my rage-edit. IndexError should be already handled by local_to_global